tests/Microsoft.Experimental.Collections.Tests/Microsoft/Collections/Extensions/OrderedDictionaryTests.cs

```diff
@@ -962,5 +963,102 @@ public void TestLargeOrderedDictionaryOperations()
 
             Assert.False(dictionary.ContainsKey(count.ToString()));
         }
+
+        [Fact]
+        public void RemoveSlotReused()
+        {
+            OrderedDictionary<Collider, int> d = new OrderedDictionary<Collider, int>();
+            d[C(0)] = 0;
+            d[C(1)] = 1;
+            d[C(2)] = 2;
+            Assert.True(d.Remove(C(0)));
+
+            d[C(0)] = 3;
+            Assert.Equal(d[C(0)], 3);
+            Assert.Equal(3, d.Count);
+        }
+
+        [Fact]
+        public void OrderedDictionaryVersusDictionary_AllCollisions()
+        {
+            Random rand = new Random(333);
+            OrderedDictionary<Collider, int> od = new OrderedDictionary<Collider, int>();
+            Dictionary<Collider, int> d = new Dictionary<Collider, int>();
+            int size = rand.Next(1234);
+            for (int i = 0; i < size; i++)
+            {
+                if (rand.Next(5) != 0)
+                {
+                    Collider k = C(rand.Next(100) + 23);
+                    int v = rand.Next();
+                    if (od.TryGetValue(k, out int t))
+                    {
+                        od[k] = t + v;
+                    }
+                    else
+                    {
+                        od[k] = v;
+                    }
+                    if (d.TryGetValue(k, out t))
+                    {
+                        d[k] = t + v;
+                    }
+                    else
+                    {
+                        d[k] = v;
+                    }
+                }
+                if (rand.Next(3) == 0 && d.Count > 0)
+                {
+                    Collider el = GetRandomElement(d);
+                    Assert.True(od.Remove(el));
+                    Assert.True(d.Remove(el));
+                }
+            }
+            Assert.Equal(d.Count, od.Count);
+            Assert.Equal(d.OrderBy(i => i.Key), (od.OrderBy(i => i.Key)));
+            Assert.Equal(d.OrderBy(i => i.Value), (od.OrderBy(i => i.Value)));
+        }
+
+        private TKey GetRandomElement<TKey, TValue>(IDictionary<TKey, TValue> d)
+        {
+            int index = 0;
+            Random rand = new Random(42);
+            foreach (KeyValuePair<TKey, TValue> entry in d)
+            {
+                if (rand.Next(d.Count) == 0 || index == d.Count - 1)
+                {
+                    return entry.Key;
+                }
+                index++;
+            }
+            throw new InvalidOperationException();
+        }
+
+        [DebuggerStepThrough]
+        internal Collider C(int val) => new Collider(val);
     }
 }
+
+[DebuggerDisplay("{Key}")]
+internal struct Collider : IEquatable<Collider>, IComparable<Collider>
```

Comment:
Any reason why this is not part of the tests namespace?

Reply:
I just mimicked what @danmosemsft did in the DictionarySlim PR. I can pull it into the namespace or perhaps as a nested type.